src/Core/Framework/DataAbstractionLayer/DataAbstractionLayerException.php

```diff
@@ -1006,4 +985,14 @@ public static function unexpectedConstraintType(Constraint $constraint, string $
     {
         return new UnexpectedTypeException($constraint, $expectedType);
     }
+
+    public static function invalidCriteriaParameter(string $message): self
```

Comment:
```suggestion
    public static function invalidCompressedCriteriaParameter(string $message): self
```

and let's adjust the const as well to make it clear that is for the `_criteria` param and not for general criteria errors, as that distinction is currently hard to make